gateway-ha/src/main/java/io/trino/gateway/baseapp/BaseApp.java

```diff
@@ -68,6 +68,9 @@ public abstract class BaseApp
 {
     private static final Logger logger = Logger.get(BaseApp.class);
     private final ImmutableList.Builder<Module> appModules = ImmutableList.builder();
+    // this injector reference is needed to use reflection in
+    // TestGatewayHaSingleBackend and TestGatewayMultipleBackend
```

Comment:
Please avoid using reflection. We should find different ways.

Reply:
if we don't use reflection another way is to add a timeout to let healthcheck finish checking the health since `TestGatewayHaSingleBackend` and `TestGatewayMultipleBackend` are integration tests